### PR TITLE
[feature](vault) Support alter storage vault stmt in FE

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Support alter storage vault stmt in FE. Currently only support alter s3 vault's ak and sk.
For example
```sql
alter storage vault s3_vault properties(
"type" = "s3",
"ak" = "new_ak",
"sk" = "new_sk"
);
```

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 1]
```diff
@@ -509,56 +510,182 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
...
+        msg = fmt::format("failed to commit kv txn, err={}", err);
+        LOG(WARNING) << msg;
+    }
+
+    return 0;
+}
+
+static int extract_object_storage_info(
+        const AlterObjStoreInfoRequest* request, MetaServiceCode& code, std::string& msg,
+        std::tuple<std::string&, std::string&, std::string&, std::string&, std::string&,
```
Comment (platoneko) [inline cloud/src/meta-service/meta_service_resource.cpp:589]: Consider defining a struct.

**Review by platoneko — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Review by platoneko — DISMISSED** [review]

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 2]
```diff
@@ -509,56 +510,193 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
...
+        return -1;
+    }
+    const auto& name = vault.name();
+    auto name_itr = std::find_if(instance.storage_vault_names().begin(),
+                                 instance.storage_vault_names().end(),
+                                 [&](const auto& vault_name) { return name == vault_name; });
+    if (name_itr == instance.storage_vault_names().end()) {
+        code = MetaServiceCode::INVALID_ARGUMENT;
+        std::stringstream ss;
+        ss << "invalid storage vault name, name =" << name;
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:539]: change to "invalid storage vault name, not found, name="

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 3]
```diff
@@ -865,6 +951,201 @@ void MetaServiceImpl::alter_obj_store_info(google::protobuf::RpcController* cont
...
+        if (0 != extract_object_storage_info(request, code, msg, tmp_desc, encryption_info,
+                                             cipher_ak_sk_pair)) {
+            return;
+        }
+    } break;
+    case AlterObjStoreInfoRequest::ADD_BUILT_IN_VAULT: {
+        // It should at least has one hdfs info or obj info inside storage vault
+        if ((!request->has_vault())) {
+            code = MetaServiceCode::INVALID_ARGUMENT;
+            msg = "hdfs info is not found " + proto_to_json(*request);
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:977]: typo hdfs?

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 4]
```diff
@@ -865,6 +951,201 @@ void MetaServiceImpl::alter_obj_store_info(google::protobuf::RpcController* cont
...
+    case AlterObjStoreInfoRequest::LEGACY_UPDATE_AK_SK:
+    case AlterObjStoreInfoRequest::UPDATE_AK_SK: {
+        auto tmp_desc =
+                ObjectStorageDesc {ak, sk, bucket, prefix, endpoint, external_endpoint, region};
+        if (0 != extract_object_storage_info(request, code, msg, tmp_desc, encryption_info,
+                                             cipher_ak_sk_pair)) {
+            return;
+        }
+    } break;
+    case AlterObjStoreInfoRequest::ADD_BUILT_IN_VAULT: {
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:973]: is this necessary for the lagacy `object_store` backends?

File: cloud/src/meta-service/meta_service_resource.cpp [hunk 5]
```diff
@@ -509,56 +510,193 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
...
+    }
+    auto pos = name_itr - instance.storage_vault_names().begin();
+    auto id_itr = instance.resource_ids().begin() + pos;
+    auto vault_key = storage_vault_key({instance.instance_id(), *id_itr});
+    std::string val;
+
+    auto err = txn->get(vault_key, &val);
+    LOG(INFO) << "get instance_key=" << hex(vault_key);
+
+    if (err != TxnErrorCode::TXN_OK) {
```
Comment (gavinchou) [inline cloud/src/meta-service/meta_service_resource.cpp:551]: print the obj info before and after change.
mask the sk by replacing the middle chars.
e.g 
```
sk = abcdefgh
print abc***fgh 
```

**Review by gavinchou — CHANGES_REQUESTED** [review]
Pls add  regression test for the new feature

**Comment by ByteYue** [comment]
run feut

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by Yukang-Lian — APPROVED** [review]

**Pull request merged**